core/database/src/main/java/com/gkp/core/database/mapper/Mapper.kt

```diff
@@ -0,0 +1,74 @@
+package com.gkp.core.database.mapper
+
+import com.gkp.agenda.domain.model.AgendaItem
+import com.gkp.core.database.entity.AgendaItemEntity
+
+fun AgendaItemEntity.toAgendaItem(): AgendaItem {
+    return when {
+        isDone != null -> {
+            AgendaItem.Task(
+                id = id,
+                title = title,
+                description = description,
+                time = time,
+                remindAt = remindAt,
+                isDone = isDone
+            )
+        }
+        to != null -> {
```

Comment:
This isn’t a very reliable way to check the type of agenda item here 😅
Check my comment on having the items as separate entities.